Intersect.Client/Entities/Entity.cs

```diff
@@ -856,7 +856,71 @@ public virtual bool IsStealthed
         /// <summary>
         /// Returns whether the name of this entity should be drawn.
         /// </summary>
-        public virtual bool ShouldDrawName => !HideName && ShouldDraw;
+        public virtual bool ShouldDrawName
+        {
+            get
+            {
+                // Return if the map instance is null or the name is empty or not set to be drawn.
+                if (MapInstance == null || string.IsNullOrWhiteSpace(Name) || HideName || !ShouldDraw)
+                {
+                    return false;
+                }
+
+                // Look up the mouse position and convert it to a world point.
+                var mousePos = Graphics.ConvertToWorldPoint(Globals.InputManager.MousePosition);
+
+                // Entity is considered hovered if the mouse is over its world position and not hovering over the GUI.
+                IsHovered = WorldPos.Contains(mousePos.X, mousePos.Y) && !Interface.Interface.MouseHitGui();
+
+                // Check the type of entity and return whether its name should be drawn based on settings and conditions.
+                switch (this)
+                {
+                    case Projectile _:
+                    case Resource _:
+                        return false;
+                    case Event _:
+                        return true;
+                    case Player player:
+                        if (IsHovered)
+                        {
+                            return true;
+                        }
+
+                        var me = Globals.Me;
+                        bool meetsRequirements = false;
+
+                        if (Globals.Database.MyOverheadInfo && player.Id == me.Id)
+                        {
+                            meetsRequirements = true;
+                        }
+                        else if (Globals.Database.PlayerOverheadInfo && player.Id != me.Id)
+                        {
+                            meetsRequirements = true;
+                        }
+                        else if (Globals.Database.PartyMemberOverheadInfo && me.IsInMyParty(player))
+                        {
+                            meetsRequirements = true;
+                        }
+                        else if (Globals.Database.FriendOverheadInfo && me.IsFriend(player))
+                        {
+                            meetsRequirements = true;
+                        }
+                        else if (Globals.Database.GuildMemberOverheadInfo && me.IsGuildMate(player))
+                        {
+                            meetsRequirements = true;
+                        }
+
+                        return meetsRequirements;
```

Comment:
Instead of the variable and `else`s, just do returns

Also add a blank line between all of the returns and the case after them, it makes the cases blend together without the vertical white space